src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,476 @@
+        const func = node.arguments[0];
+        const argName = getArgName(func, synthNamesMap, checker);
+
+        let varDecl;
+        if (prevArgName && lastDotThenMap.get(getNodeId(node).toString())) {
+            varDecl = createVariableStatement(/*modifiers*/ undefined, createVariableDeclarationList([createVariableDeclaration(getSynthesizedDeepClone(prevArgName[0]))], NodeFlags.Let /*getNodeFlags(prevArgName[0], constIdentifiers)*/));
+            prevArgName[1] += 2;
+        }
+        const tryBlock = createBlock(parseCallback(node.expression, checker, node, synthNamesMap, lastDotThenMap, context, constIdentifiers, prevArgName));
+
+        const callbackBody = getCallbackBody(func, prevArgName, argName, node, checker, synthNamesMap, lastDotThenMap, context, constIdentifiers);
+        const catchClause = createCatchClause(argName[0].text, createBlock(callbackBody));
+
+        const tryStatement = createTry(tryBlock, catchClause, /*finallyBlock*/ undefined);
+        return varDecl ? [varDecl, tryStatement] : [tryStatement];
+    }
+
+    function parseThen(node: CallExpression, checker: TypeChecker, outermostParent: CallExpression, synthNamesMap: Map<[Identifier, number]>,
+        lastDotThenMap: Map<boolean>, context: CodeFixContextBase, constIdentifiers: Identifier[], prevArgName?: [Identifier, number]): Statement[] {
+
+        const [res, rej] = node.arguments;
+
+        if (!res) {
+            return parseCallback(node.expression, checker, outermostParent, synthNamesMap, lastDotThenMap, context, constIdentifiers, prevArgName);
+        }
+
+        const argNameRes = getArgName(res, synthNamesMap, checker);
+        const callbackBody = getCallbackBody(res, prevArgName, argNameRes, node, checker, synthNamesMap, lastDotThenMap, context, constIdentifiers);
+
+        if (rej) {
+            const argNameRej = getArgName(rej, synthNamesMap, checker);
+
+            const tryBlock = createBlock(parseCallback(node.expression, checker, node, synthNamesMap, lastDotThenMap, context, constIdentifiers, argNameRes).concat(callbackBody));
+
+            const callbackBody2 = getCallbackBody(rej, prevArgName, argNameRej, node, checker, synthNamesMap, lastDotThenMap, context, constIdentifiers);
+            const catchClause = createCatchClause(argNameRej[0].text, createBlock(callbackBody2));
+
+            return [createTry(tryBlock, catchClause, /*finallyBlock*/ undefined) as Statement];
+        }
+        else {
+            return parseCallback(node.expression, checker, node, synthNamesMap, lastDotThenMap, context, constIdentifiers, argNameRes).concat(callbackBody);
+        }
+
+        return [];
+    }
+
+    function getFlagOfIdentifier(node: Identifier, constIdentifiers: Identifier[]): NodeFlags {
+        const inArr: boolean = constIdentifiers.filter(elem => elem.text === node.text).length > 0;

```

Comment:
If you use `some`, instead of `filter` and `length`, it can kick out as soon as it finds one.